src/managers/SubscriptionManager.py

```diff
@@ -49,3 +49,179 @@ def update_subscription(self, subscription_id, subscription_update_dto):
         except Exception as e:
             logging.error(f"Error updating subscription: {e}")
             raise
+
+    def delete_subscription(self, subscription_id):
+        try:
+            return self.subscription_store.delete_subscription(subscription_id)
+        except OperationalError as e:
+            logging.error(f"Database error deleting subscription: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error deleting subscription: {e}")
+            raise
+
+    def activate_subscription(self, subscription_id):
+        try:
+            subscription = self.get_subscription(subscription_id)
+            if subscription:
+                subscription.activate()
+                return self.subscription_store.update_subscription(subscription_id, subscription)
+            else:
+                raise ValueError("Subscription not found")
+        except OperationalError as e:
+            logging.error(f"Database error activating subscription: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error activating subscription: {e}")
+            raise
+
+    def cancel_subscription(self, subscription_id):
+        try:
+            subscription = self.get_subscription(subscription_id)
+            if subscription:
+                subscription.cancel()
+                return self.subscription_store.update_subscription(subscription_id, subscription)
+            else:
+                raise ValueError("Subscription not found")
+        except OperationalError as e:
+            logging.error(f"Database error cancelling subscription: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error cancelling subscription: {e}")
+            raise
+
+    def expire_subscription(self, subscription_id):
+        try:
+            subscription = self.get_subscription(subscription_id)
+            if subscription:
+                subscription.expire()
+                return self.subscription_store.update_subscription(subscription_id, subscription)
+            else:
+                raise ValueError("Subscription not found")
+        except OperationalError as e:
+            logging.error(f"Database error expiring subscription: {e}")
+            raise
+        except ProgrammingError as e:
+            logging.error(f"Invalid SQL query: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error expiring subscription: {e}")
+            raise
+
+    def update_subscription_status(self, subscription_id):
```

Comment:
This method needs to receive a status as an argument. Pass the status to the update_status(status) method